src/components/Market/MarketFilter/index.tsx

```diff
@@ -0,0 +1,313 @@
+  };
+
+  const clearFilters = () => {
+    setStartDate(defaultFilterState.startDate);
+    setEndDate(defaultFilterState.endDate);
+    setOccupancyIndex(defaultFilterState.occupancyIndex);
+    setDurationIndex(defaultFilterState.durationIndex);
+    setPriceRange(defaultFilterState.priceRange);
+
+    setFilter(defaultFilterState);
+  };
+
+  const applyFilters = () => {
+    /** Apply filters */
+    setFilter({
+      startDate,
+      endDate,
+      occupancyIndex,
+      durationIndex,
+      priceRange,
+    } as FilterState);
+
+    const unit = Math.pow(10, decimals);
+
+    onChange({
+      startDate,
+      endDate,
+      coreOccupancy: coreOccupancyOptions[occupancyIndex].limit,
+      minDuration: durationOptions[durationIndex].duration,
+      price:
+        priceRange === undefined
+          ? undefined
+          : {
+              min: new BN(priceRange[0] * unit),
+              max: new BN(priceRange[1] * unit),
+            },
+    } as MarketFilterOptions);
+  };
+
+  return (
+    <Box>
+      <Button
+        aria-describedby={id}
+        onClick={(e) => openFilters(e)}
+        sx={{
+          background: open
+            ? theme.palette.common.black
+            : theme.palette.common.white,
+          color: open ? theme.palette.common.white : theme.palette.common.black,
+          borderRadius: '2rem',
+          px: '1.5rem',
+          height: '100%',
+          fontSize: '1rem',
+          fontWeight: 400,
+        }}
+        endIcon={<SettingsInputComponentRoundedIcon />}
+      >
+        Filters
+      </Button>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        onClose={() => closeFilters()}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right',
+        }}
+      >
+        <Paper sx={{ padding: '1.5rem', width: '20rem' }}>
+          <Stack
+            direction='row'
+            justifyContent='space-between'
+            alignItems='center'
+          >
+            <Typography
+              sx={{
+                fontSize: '1rem',
+                fontWeight: 700,
+                color: theme.palette.common.black,
+              }}
+            >
+              Filters
+            </Typography>
+            <Button
+              startIcon={<CloseOutlinedIcon />}
+              sx={{
+                color: theme.palette.common.black,
+                display: 'flex',
+                alignItems: 'center',
+              }}
+              onClick={() => clearFilters()}
+            >
+              Clear all
+            </Button>
+          </Stack>
+          <Stack direction='column' gap='1rem' sx={{ my: '1rem' }}>
+            <FilterItem label='Start date'>
+              <DateInput value={startDate} onChange={(v) => setStartDate(v)} />
+            </FilterItem>
+            <FilterItem label='End date'>
```

Comment:
Currently, the end date filter shows regions ending before the selected date. In my opinion, it would make more sense to filter all the regions that end after the selected date.

For example, if I need a region from Monday to Friday, I would rather see options that end later than Friday, as options that end sooner won't meet my needs.

WDYT?

Reply:
That sounds good.